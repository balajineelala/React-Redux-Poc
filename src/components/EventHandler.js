import React, { Component } from "react";

class EventHandler extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: "Initial data on component load...",
    };
    this.updateState = this.updateState.bind(this);
  }
  updateState() {
    this.setState({ data: "Data updated after event triggering..." });
  }
  render() {
    const { buttonName } = this.props;
    return (
      <div>
        <h4>{this.state.data}</h4>
        <button onClick={this.updateState}>{buttonName}</button>
      </div>
    );
  }
}
export default EventHandler;
