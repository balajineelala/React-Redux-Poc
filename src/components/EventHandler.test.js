import React from "react";
import { shallow, mount } from "enzyme";
import EventHandler from "./EventHandler";

describe("EventHandler component", () => {
  //find button in the component
  it("Expects to find button HTML element in the DOM", () => {
    const wrapper = mount(<EventHandler buttonName="test button" />);
    expect(wrapper.find("button")).toHaveLength(1);
  });

  //simulate button click event
  it("simulate button click event", () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <EventHandler onClick={mockCallBack}>Test Update</EventHandler>
    );
    wrapper.find("button").simulate("click");
    const text = wrapper.find("h4").text();
    expect(text).toEqual("Data updated after event triggering...");
  });

  //method called or not generating snapshot
  it("updateState method as expected", () => {
    const component = shallow(<EventHandler />);
    const classInstance = component.instance();
    classInstance.updateState();
    const newState = classInstance.state.data;
    expect(newState).not.toBe(null);
    expect(component).toMatchSnapshot();
  });
});
