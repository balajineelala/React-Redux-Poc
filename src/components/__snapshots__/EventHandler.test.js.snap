// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHandler component updateState method as expected 1`] = `ShallowWrapper {}`;
